docs/tutorials_generator.py

```diff
@@ -0,0 +1,22 @@
+import os
+
+notebook_list = []
+
+walk = os.walk("io/")
+
+for pth, dirs, files in walk:
+    for file in files:
+        if ".ipynb" in file and "checkpoint" not in file:
```

Comment:
Try using method `endswith` for strings and `isdir` for checkpoint directories. Also we use a lot `fname` instead of `file` throughout TARDIS code.